presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1068,14 +1204,15 @@ public synchronized void cleanupQuery(ConnectorSession session)
         HiveIdentity identity = new HiveIdentity(session);
         checkState(currentQueryId.equals(Optional.of(queryId)), "Invalid query id %s while current query is", queryId, currentQueryId);
         Optional<HiveTransaction> transaction = currentHiveTransaction;
-        currentQueryId = Optional.empty();
-        currentHiveTransaction = Optional.empty();
-        hiveTransactionSupplier = Optional.empty();
 
         if (transaction.isEmpty()) {
+            clearCurrentTransaction();
             return;
         }
 
+        commit();
```

Comment:
What's the reason to call commit within cleanupQuery?
what if 

- query is part of a longer transaction (we seem to commit everything too early)
- query is cancelled/aborted/etc -- it seems we commit instead of rolling back

please help me understand

cc @losipiuk